### DISCUSSION TITLE
Disabled site on the front end is still visible

### DISCUSSION THREAD

**Discussion by yingban** [craftcms/cms#9044] [Ideas]
### Description

HI,

As discussed with Oli by email, I disabled a secondary website in a multisite environment, and it is stil accessible on the front end, also when logout.

Regarding Oli's answer, it should better return a 503.

Thank you the support.

### System
Craft CMS 3.6.14

**Comment by brandonkelly** [comment]
To be clear, disabled sites are not accessible; Craft just completely ignores them during routing, as if they doesn’t exist.

When a request comes in, Craft compares the incoming URL to each of the (enabled) sites’ base URLs. If an exact match is found, that site will be served. Otherwise it will default to the primary site, which is what’s happening in your case.

That said, we could theoretically start including disabled sites when comparing the requested URL to sites’ base URLs, and if there’s an exact match but the site turns out to be disabled, respond with a 404 or 503 status.

Or perhaps we could add a config setting that starts responding with a 404 status any time an exact site match is not found.

**Comment by ademers** [comment]
I'd like to be able to take individual sites offline, rather than the entire system. For example, I have a multi-site install with a private (members-only, login required) site and a public (marketing) site. I do regular maintenance mostly on the private site (more complex features), and would really like to be able to keep the public site online, while showing my offline/503 template on the private site only. Ideally, I'd like to be able to take sites offline/online via config setting and/or Craft CLI command. May something like `craft on/off` with a site parameter? Thanks!

**Comment by brandonkelly** [comment]
@ademers To be clear it’s already possible to take individual sites offline, as of Craft 3.5 per #3005, and how that works (project config change) isn’t really up for debate here; just what should happen when a disabled site is requested.

**Comment by ademers** [comment]
@ademers Thanks for clarifying. I completely missed the fact that this has been possible since Craft 3.5. So, to answer your question, I'd like visitors to see a custom 503/offline template.

**Comment by yingban** [comment]
Thanks for your replies, same as @ademers 503 sounds good.

**Comment by rob-c-baker** [comment]
IMO if a disabled site is requested I can see arguments for `403` _(user not allowed to see the site)_ or `404` _(the site is not found)_ status codes but I think on balance it's in a different state than those. `503` seems the next most reasonable, even though the implication is that it's a temporary situation - it might not be.

**Comment by brandonkelly** [comment]
It’s not possible to disable a site for some people but not others (the _only_ exception being if you are previewing an entry in a disabled site w/ Live Preview/Share), so 403 feels wrong. 404 would be relatively consistent with the current behavior (nothing to see here!) if applied to all non-exact-matches; 503 would make sense if the existence of the site should be known, and it’s OK to reveal that it’s just not ready for prime time.
